roxie/ccd/ccdserver.cpp

```diff
@@ -11273,10 +11270,25 @@ class CRoxieServerXmlWriteActivity : public CRoxieServerDiskWriteActivity
         desc->queryProperties().setProp("@format","utf8n");
         desc->queryProperties().setProp("@rowTag",rowTag.get());
         desc->queryProperties().setProp("@kind", (kind==TAKjsonwrite) ? "json" : "xml");
-        if (headerLength)
-            desc->queryProperties().setPropInt("@headerLength", headerLength);
-        if (footerLength)
-            desc->queryProperties().setPropInt("@footerLength", footerLength);
+        if (kind==TAKjsonwrite)
+        {
+            if (headerLength)
+                desc->queryProperties().setPropInt("@headerLength", headerLength);
+            if (footerLength)
+                desc->queryProperties().setPropInt("@footerLength", footerLength);
+        }
+        else
+        {
+            IPartDescriptorIterator *parts = desc->getIterator();
+            if (parts->first())
+            {
+                do {
+                    IPartDescriptor &part = parts->query();
+                    part.queryProperties().setPropInt("@headerLength", headerLength);
+                    part.queryProperties().setPropInt("@footerLength", footerLength);
```

Comment:
Similar to hthor comment:
it may make sense, but could do with a comment here to explain why json does not set @headerLength/@footerLength if 0, but xml does.